packages/terra-clinical-data-grid/src/DataGrid.jsx

```diff
@@ -0,0 +1,1200 @@
+    this.setScrollbarRef = this.setScrollbarRef.bind(this);
+    this.setScrollbarContainerRef = this.setScrollbarContainerRef.bind(this);
+    this.setTerminalFocusAnchorRef = this.setTerminalFocusAnchorRef.bind(this);
+    this.setVerticalOverflowContainerRef = this.setVerticalOverflowContainerRef.bind(this);
+    this.cellRefs = {};
+    this.headerCellRefs = {};
+    this.sectionRefs = {};
+
+    /**
+     * Resize Events
+     */
+    this.handleDataGridResize = this.handleDataGridResize.bind(this);
+    this.resizeSectionHeaders = this.resizeSectionHeaders.bind(this);
+    this.updateHeaderScrollbarBuffer = this.updateHeaderScrollbarBuffer.bind(this);
+
+    /**
+     * Scroll synchronization
+     */
+    this.synchronizeHeaderScroll = this.synchronizeHeaderScroll.bind(this);
+    this.synchronizeContentScroll = this.synchronizeContentScroll.bind(this);
+    this.synchronizeScrollbar = this.synchronizeScrollbar.bind(this);
+    this.resetHeaderScrollEventMarkers = this.resetHeaderScrollEventMarkers.bind(this);
+    this.resetContentScrollEventMarkers = this.resetContentScrollEventMarkers.bind(this);
+    this.resetScrollbarEventMarkers = this.resetScrollbarEventMarkers.bind(this);
+    this.updateScrollbarPosition = this.updateScrollbarPosition.bind(this);
+    this.updateScrollbarVisibility = this.updateScrollbarVisibility.bind(this);
+    this.scrollbarPosition = 0;
+
+    /**
+     * Rendering
+     */
+    this.renderCell = this.renderCell.bind(this);
+    this.renderHeaderCell = this.renderHeaderCell.bind(this);
+    this.renderRowSelectionCell = this.renderRowSelectionCell.bind(this);
+    this.renderHeaderRow = this.renderHeaderRow.bind(this);
+    this.renderOverflowContent = this.renderOverflowContent.bind(this);
+    this.renderPinnedContent = this.renderPinnedContent.bind(this);
+    this.renderRow = this.renderRow.bind(this);
+    this.renderScrollbar = this.renderScrollbar.bind(this);
+    this.renderSection = this.renderSection.bind(this);
+    this.renderSectionHeader = this.renderSectionHeader.bind(this);
+
+    /**
+     * Determining the widths of the pinned and overflow sections requires iterating over the prop arrays. The widths are
+     * generated and cached in state to limit the amount of iteration performed by the render functions.
+     */
+    this.state = {
+      pinnedColumnWidth: DataGrid.getTotalColumnWidth(DataGrid.getPinnedColumns(props), props.defaultColumnWidth),
+      overflowColumnWidth: DataGrid.getTotalColumnWidth(DataGrid.getOverflowColumns(props), props.defaultColumnWidth),
+    };
+  }
+
+  componentDidMount() {
+    /**
+     * A ResizeObserver is used to manage changes to the DataGrid's overall size. The handler will execute once upon the start of
+     * observation and on every subsequent resize.
+     */
+    this.resizeObserver = new ResizeObserver((entries) => { this.handleDataGridResize(entries[0].contentRect.width, entries[0].contentRect.height); });
```

Comment:
I'm sure I am missing it since this file is pretty large. I see the Resize Observer on Mount and Dismount, where do else is it working it's magic?